### PR TITLE
Add option 'leave_visible'

### PR DESCRIPTION
**Pull request by prince-chrismc**
to configure the number of comments to leave visible

❓ Test Code? Currently the way it's written it's no very easy to test, if you'd like I have a few suggestions on how it can be done.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
Its failing for this https://github.com/prince-chrismc/label-merge-conflicts-action#faq---how-do-i-fix-resource-not-accessible-by-integration

File: action.yml [hunk 1]
```diff
@@ -9,13 +9,17 @@ inputs:
...
+    description: 'GitHub username to hide comments'
     default: 'github-actions[bot]'
     required: false
   hide_reason:
     description: 'Reasons to hide comments'
     default: 'OUTDATED'
     required: false
+  leave_visible:
+    description: 'The number of most recent comments to leave untouched'
+    default: '1'
```
Comment (kanga333) [inline action.yml]:
```suggestion
    default: '0'
```

I would like the default to be 0 as I want the same behaviour as existing.

File: src/main.ts [hunk 2]
```diff
@@ -6,10 +6,11 @@ async function run(): Promise<void> {
     const token: string = core.getInput('github_token')
     const userName: string = core.getInput('hide_user_name')
     const reason: string = core.getInput('hide_reason')
+    const nbOfCommentsToLeave: number = parseInt(core.getInput('leave_visible'), 10)
 
     const cli = new Client(token)
     const ids = await cli.SelectComments(userName)
-    for (const id of ids) {
+    for (const id of ids.splice(-1, nbOfCommentsToLeave)) {
```
Comment (kanga333) [inline src/main.ts]:
```suggestion
    for (const id of ids.splice(-nbOfCommentsToLeave, nbOfCommentsToLeave)) {
```

In order to remove an arbitrary number of elements from the end of the array, I think start also needs to be moved back a `nbOfCommentsToLeave`.

File: action.yml [hunk 3]
```diff
@@ -9,13 +9,17 @@ inputs:
     description: 'The GitHub token used to create an authenticated client'
     required: true
   hide_user_name:
-    description: 'Github username to hide comments'
+    description: 'GitHub username to hide comments'
```
Comment (kanga333) [inline action.yml:12]: Good catch! thanks.

**Review by kanga333 — COMMENTED** [review]
You can ignore the test this time.

I would like to add an integration test using Personal Access Token to replace the currently failing test.
At that time I will add a test case using `leave_visible`.

File: action.yml (hunk 1 above)
Comment (prince-chrismc) [reply action.yml]: That's very acceptable

File: src/main.ts (hunk 2 above)
Comment (prince-chrismc) [reply src/main.ts]: You are absolutely correct!

**Comment by kanga333** [comment]
Thanks for the PR. I will release this change as v0.2.0.

**Comment by kanga333** [comment]
@prince-chrismc Hi, I found a bug in the behaviour of leave_visible and fixed it in #8.
Please use the corrected version, [v0.2.1](https://github.com/kanga333/comment-hider/releases/tag/v0.2.1). Thanks.

**Pull request merged**